### DISCUSSION TITLE
Signature Images

### DISCUSSION THREAD

**Discussion by ChrisNolan** [karrioapi/karrio#689] [General]
So I've been working to update the 'tracker' display page, and wanted to get the 'signature image' if it was available to be linked and displayable etc.

I'm feeling like maybe all the work with POD (Proof of Delivery) images hasn't been fully ironed out yet though?  Not sure.

Seems like it was originally added in Pull Request #501 which support for UPS originally added.  Then commit 7f532343518a9f93e863a334b7daf11286c66c9b Fedex support was added.

I found some bugs in that , and have worked through them so now that signature image is being stored in the 'tracking-status' table correctly :tada: but then I went to 'look' at the image on the dashboard and nope... trouble.

So the .signature_image_url seems to just be a function which returns a url of the form "http://localhost:3000/v1/trackers/trk_b5b3611878ca4e3584ce214096975842/signature_image" -- but this 404s (oh btw, I've also made some changes because it seems like 404s were not being displayed at all).  I couldn't find anything in the dashboard's pages or api folders that would match this URL.  So... I went to make it... but while I was struggling with getting the folder structure correct, I figured ok I'll just display the base64 string directly in an img tag in the mean time just so I can have a satisfactory conclusion to my afternoon... but that just doesn't seem to be exposed or something?

I see that maybe it is supposed to be on tracker.images but that is empty... but images is part of TrackerDetails and not TrackerStatus as a type... and... well I'm tired and my brain is melted a bit so I'm going to step away but I'd love it if others who have worked with their POD images successfully can share what they did, or what I'm missing, or where I can go to keep adding to their growing branching.

Oh, if you want to see the code... I believe you can see my work in progress @ ... https://github.com/saunders-book-co/karrio/commits/fedex_tracker_updates/ if that'll help at all.

Oh, and on a connected note -- I'm pretty sure from what I can see FedEx doesn't offer the Photo Proof of Delivery (delivery image) in their API at this point, is that what you all can see as well? It is on their website, but  not the API?

Cheers!

**Comment by ChrisNolan** [comment]
Any ideas about how to access the 'signature_image' from the dashboard app?  Suggestions on how to implement it if it isn't already implemented?  I feel like I'm 'close' but still missing a piece of the puzzle.

**Comment by danh91** [comment]
Hi @ChrisNolan ,

I am catching up with the issues and missed this one.

Do you mind opening a PR for the changes you are doing related to this issue? It will make it easier to review 

Thanks

**Reply by ChrisNolan** [reply]
Ah... sure, but in my brain shouldn't the pull request be 'finished' before I make it...?  It feels like I'm only '3/4' of the way done... but I'll give it a shot to see...

**Reply by ChrisNolan** [reply]
Ok, that is there now as pull #692 (always anxious when others see my code 😉 )
I didn't fix conflicts yet or anything.
